packages/wrangler/src/config-cache.ts

```diff
@@ -1,44 +1,82 @@
 import { mkdirSync, readFileSync, rmSync, writeFileSync } from "fs";
-import { dirname, join } from "path";
+import * as path from "path";
+import { findUpSync } from "find-up";
+import { logger } from "./logger";
 
 let cacheMessageShown = false;
 
-const cacheFolder = "node_modules/.cache/wrangler";
+let __cacheFolder: string | null;
+function getCacheFolder() {
+	if (__cacheFolder || __cacheFolder === null) return __cacheFolder;
 
-const showCacheMessage = () => {
+	const closestNodeModulesDirectory = findUpSync("node_modules", {
```

Comment:
I guess one approach would be for the developer to create `.env` files for each Worker directory and put `CLOUDFLARE_ACCOUNT_ID` variables there?

Reply:
This situation already happens in our own repo; running `npx wrangler dev` inside `fixtures/worker-app` stores the cached value under the root `node_modules`. I think this is fine, it's very rare that you'd be deploying to different account IDs in the same repo. And in that situation, yo'ujd likely do as you mentioned: use a .env file with an account id, and/or some other scripting.

That's also why it's important to log to terminal when we fetch the account id, so that a user knows where to go and blow it away. I think we may have to introduce a command to purge this cache as well, and we'll do that later.